src/librustc_data_structures/profiling.rs

```diff
@@ -216,34 +224,35 @@ impl SelfProfilerRef {
     /// VerboseTimingGuard returned from this call is dropped. In addition to recording
     /// a measureme event, "verbose" generic activities also print a timing entry to
     /// stdout if the compiler is invoked with -Ztime or -Ztime-passes.
-    #[inline(always)]
     pub fn verbose_generic_activity<'a>(
         &'a self,
         event_id: &'static str,
     ) -> VerboseTimingGuard<'a> {
-        VerboseTimingGuard::start(
-            event_id,
-            self.print_verbose_generic_activities,
-            self.generic_activity(event_id),
-        )
+        let message =
+            if self.print_verbose_generic_activities { Some(event_id.to_owned()) } else { None };
```

Comment:
It could be turned into a `Cow`, yes. But I left it as `'static` because that's a good safeguard against accidentally encoding some kind of parameter in the event label (for which purpose one should use `generic_activity_with_arg` or `extra_verbose_generic_activity`). Let me rename `event_id` to `event_label` in the cases where we don't support parameters...

Reply:
I updated the names.